components/Msig/Withdraw.tsx

```diff
@@ -0,0 +1,218 @@
+  const { loginOption, walletProvider, walletError, getProvider } =
+    useWalletProvider()
+
+  // Input states
+  const [toAddress, setToAddress] = useState<string>('')
+  const [value, setValue] = useState<FilecoinNumber | null>(null)
+  const [isToAddressValid, setIsToAddressValid] = useState<boolean>(false)
+  const [isValueValid, setIsValueValid] = useState<boolean>(false)
+
+  // Transaction states
+  const [txState, setTxState] = useState<TxState>(TxState.FillingForm)
+  const [txError, setTxError] = useState<Error | null>(null)
+
+  // Params to pass with the withdraw message
+  const [params, setParams] = useState<WithdrawParams | null>(null)
+
+  // Prevent redundant updates to params so that we don't
+  // invoke the useGetGasParams hook more than necessary
+  const setParamsIfChanged = () => {
+    if (!isToAddressValid || !isValueValid) {
+      setParams(null)
+      return
+    }
+    if (
+      !params ||
+      params.to !== toAddress ||
+      params.value !== value.toAttoFil()
+    )
+      setParams({
+        to: toAddress,
+        value: value.toAttoFil(),
+        method: MsigMethod.WITHDRAW,
+        params: ''
+      })
+  }
+
+  // Placeholder message for getting gas params
+  const message = useMemo<Message | null>(
+    () =>
+      params
+        ? new Message({
+            to: Address,
+            from: wallet.address,
+            nonce: 0,
+            value: 0,
+            method: 2,
+            params: Buffer.from(serializeParams(params), 'hex').toString(
+              'base64'
+            ),
+            gasPremium: 0,
+            gasFeeCap: 0,
+            gasLimit: 0
+          })
+        : null,
+    [params, Address, wallet.address, serializeParams]
+  )
+
+  // Max transaction fee used for getting gas params. Will be
+  // null until the user manually changes the transaction fee.
+  const [maxFee, setMaxFee] = useState<FilecoinNumber | null>(null)
+
+  // Load gas parameters when message or max fee changes
+  const {
+    gasParams,
+    loading: gasParamsLoading,
+    error: gasParamsError
+  } = useGetGasParams(walletProvider, message, maxFee)
+
+  // Calculate maximum transaction fee (fee cap times limit)
+  const calculatedFee = useMemo<FilecoinNumber | null>(() => {
+    return gasParams
+      ? getMaxGasFee(gasParams.gasFeeCap, gasParams.gasLimit)
+      : null
+  }, [gasParams])
+
+  // Attempt sending message
+  const onSend = async () => {
+    setTxState(TxState.LoadingTxDetails)
+    setTxError(null)
+    const provider = await getProvider()
+    const newMessage = new Message({
+      to: message.to,
+      from: message.from,
+      nonce: await provider.getNonce(wallet.address),
+      value: message.value,
+      method: message.method,
+      params: message.params,
+      gasPremium: gasParams.gasPremium.toAttoFil(),
+      gasFeeCap: gasParams.gasFeeCap.toAttoFil(),
+      gasLimit: new BigNumber(gasParams.gasLimit.toAttoFil()).toNumber()
+    })
+    try {
```

Comment:
Going from memory / without actually running this, I might advocate to move this `try` up so it covers `getProvider`

I think `getProvider` is liable to throw a bunch of different error types

Reply:
Just tested this in the wallet - `try` should cover `getProvider` (im making the fix over there)